controllers/k8ssandra/k8ssandracluster_controller_test.go

```diff
@@ -100,8 +100,9 @@ func TestK8ssandraCluster(t *testing.T) {
 	t.Run("CreateMultiDcClusterWithMedusa", testEnv.ControllerTest(ctx, createMultiDcClusterWithMedusa))
 	t.Run("CreateSingleDcClusterNoAuth", testEnv.ControllerTest(ctx, createSingleDcClusterNoAuth))
 	t.Run("CreateSingleDcClusterAuth", testEnv.ControllerTest(ctx, createSingleDcClusterAuth))
-	// If webhooks are installed, this testcase is handled by the webhook test
+	// If webhooks are installed, these test cases are handled by the webhook test
```

Comment:
What happens when the webhook is not installed? Do we have matching logic to prevent racks from being modified in the reconciliation loop too? Historically, we've often put validation in both places to provide more protection.

Reply:
Ah, it looks like the reconciliation logic is also protected down [here](https://github.com/k8ssandra/k8ssandra-operator/pull/709/files#diff-7e422b847100c3ee477ee61b97c8ad4da71063c1fc698d4e049081a98c89399cR535)?

Could we consolidate this logic into a single validation function (I'd suggest under `pkg/cassandra/racks/validation.go`) and call it from both the webhook and the reconciliation loop?

Reply:
> What happens when the webhook is not installed? Do we have matching logic to prevent racks from being modified in the reconciliation loop too?

Do we need that here also? cass-operator reconciliation logic would block that. Does it really matter which one fails?

Reply:
That's a fair question. I don't see why we shouldn't do the check here to minimise the number of logs the user needs to follow. If they're looking for an error in the k8ssandra-operator logs it might be confusing if it turns up in the cass-operator logs.

Reply:
We can, but I'm not sure if we're close to that "only check single logs" in reality. But that's far beyond the scope of this ticket. 

The entire webhook validation should probably be in a single method that could be called by the webhook or start of the reconcile.

Reply:
So we agree on single-method. But I also think that method should get a call in the reconciliation loop too. I'm not dogmatic about it, but it seems like a small piece of logic to provide a bit of a usability uplift - albeit in rare situations.

Reply:
I wasn't disagreeing with the reconciliation loop too if implementation is done correctly (but lets not repeat code so that the reconciliation could have different validation rules after time passes). Just saying that we have an issue where error location is already a bit unclear in many situations. cass-operator might decide to do something that k8ssandra-operator doesn't know about, so for a user that's something they need to learn to check (at least at this point).